### PR TITLE
Magiclysm: Add new potions

### PR DESCRIPTION
**Pull request by KorGgenT**
#### Summary

```SUMMARY: Mods "Magiclysm: Add new potions"```

This pr adds a couple new potions with accompanying spells:
- Cat's Grace
- Eagle's Sight
- Ogre's Strength
- Fox's Cunning

In addition, it adds the Lightning Bolt spell for stormshaper.
Also, i decided to make the spellbook itemgroup be based on 10s instead of single digit weights. I've been wanting to do this for a while, but i decided that i'll figure out the conflicts later.

### CODE REVIEW

File: data/mods/Magiclysm/Spells/magus.json [hunk 1]
```diff
@@ -198,5 +198,81 @@
...
+    "base_casting_time": 1000,
+    "difficulty": 5,
+    "flags": [ "VERBAL", "SOMATIC", "NO_LEGS", "CONCENTRATE" ],
+    "effect": "target_attack",
+    "effect_str": "eagles_sight"
+  },
+  {
+    "id": "ogres_strength",
+    "type": "SPELL",
+    "name": "Cat's Grace",
```
Comment (DenJur) [inline data/mods/Magiclysm/Spells/magus.json]:
```suggestion
    "name": "Ogre's Strength",
```

**Comment by FuelType-Memes** [comment]
You have a bunch of animals in the names and then "Ogre". Maybe "Bear's strength"? Or was it the intent? 🐻

**Comment by KorGgenT** [comment]
i feel like this is thematically appropriate for magiclysm, and sets the tone for more fantasical creatures to be added in the future.

**Comment by Contemplator11** [comment]
> i feel like this is thematically appropriate for magiclysm, and sets the tone for more fantasical creatures to be added in the future.

These names are classical from DnD. However, if I remember correctly, it was Bull's Strength. Ogre's Strength is mentioned in a classical magic item, Gauntlets of Ogre Strength.
If ogres are to be present in Magiclysm, won't it be more logical to add faerie folk first? Ogres are the result of faerie mischief with mortals.

More feedback about buff spells. As an experienced Dungeon Master, I know that an ability to stack buffs is a very balance-breaking feature. Maybe some sort of limit to stacking buffs would be appropriate? I see two alternatives: 
- Concentration mechanic, which enables a player character to concentrate on one buff spell only, and a maintain concentration check is called whenever a player character is hurt. 
- Magical contamination mechanic can be used, which fits potions greatly. It can work as radiation, causing various debilitating effects if not managed properly, such as mana loss, stats and speed reduction and even mutation at high levels.

**Comment by KorGgenT** [comment]
> If ogres are to be present in Magiclysm, won't it be more logical to add faerie folk first? Ogres are the result of faerie mischief with mortals.

I would be happy to accept more contributions.

> * Concentration mechanic, which enables a player character to concentrate on one buff spell only, and a maintain concentration check is called whenever a player character is hurt.
You know, this is a good point, and I know of one way to simulate it (though it has nothing to do with actual concentration)

**Comment by LaVeyanFiend** [comment]
I'll start adding recipes for these into my potions PR. Current ideas for ingredients are:

All: Potion Base/PB(strong alcohol base like Witcher's potion base, requires MANA_FOCUS quality from #32051 to craft), requires fire, CONCENTRATE and MANA_INFUSE quality, and alchemist or class rune(class required for Greater variants) to craft

Cat's Grace: PB(strong alcohol base like Witcher's potions, uses mana focus quality from #32051 to craft), fish, and hairball(new harvest from furred mammal corpses)
Eagle's Sight: PB, bird egg, and (down) feathers(maybe add a pair of glasses?)
Fox's Cunning: PB, brain, and paper(can't think of something "smarter" than the written word, sorry :P)
Ogre's Strength: PB, sinew, and..... something strong enemies have?

**Comment by anothersimulacrum** [comment]
> Ogre's Strength: PB, sinew, and..... something strong enemies have?

Why not add ogres?

**Comment by foulman** [comment]
> 
> 
> > Ogre's Strength: PB, sinew, and..... something strong enemies have?
> 
> Why not add ogres?

We're so far planning to add in new monsters in order of overall strength, so the goblins are coming in first (along with their own dungeon), then something like hobgoblins, orcs, bugbears, etc. An Ogre or Troll should be up there with a hulk, so they are coming, just not right now.

**Comment by Contemplator11** [comment]
> I'll start adding recipes for these into my potions PR. Current ideas for ingredients are:
> 
> All: Potion Base/PB(strong alcohol base like Witcher's potion system, requires MANA_FOCUS quality from #32051 to craft), requires fire, CONCENTRATE and MANA_INFUSE quality, and alchemist or class rune(class required for Greater variants of non-generic types) to craft
> 
> Cat's Grace: Potion Base(PB), fish, hairball(new harvest from furred mammal corpses), and..... something else representing dexterity?
> Eagle's Sight: PB, bird egg, and feathers(maybe add a pair of glasses?)
> Fox's Cunning: PB, brain, and paper(can't think of something "smarter" than the written word, sorry :P)
> Ogre's Strength: PB, sinew, and..... something strong enemies have?

Cat's Grace - maybe methamphetamine? It represents speed. =)
Eagle's Sight - maybe huckleberry/blueberry or carrots? They are good for your eyes.
Fox's Cunning - caffeine pill or tobacco? They buff intelligence in-game.
Ogre's Strength - blood from zombie brutes and hulks?

File: data/mods/Magiclysm/Spells/stormshaper.json [hunk 2]
```diff
@@ -24,6 +24,31 @@
     "difficulty": 2,
     "damage_type": "electric"
   },
+  {
+    "id": "lightning_bolt",
```
Comment (ZhilkinSerg) [inline data/mods/Magiclysm/Spells/stormshaper.json]:
```
 DEBUG    : Error: data/mods//Magiclysm/Spells/stormshaper.json: line 28:28: missing mandatory member "name"


  },
  {
    "id": "lightning_bolt",
                           ^
    "type": "SPELL",
10    "description": "The goto spell for many Stormshapers, this iconic spell does just what you expect: you shoot lightning from your fingertips.  However, this lightning is more directed than most lightning, and travels
 
 FUNCTION : bool main_menu::new_character_tab()
 FILE     : src/main_menu.cpp
 LINE     : 708
```

File: data/mods/Magiclysm/items/cast_spell_items.json [hunk 3]
```diff
@@ -60,5 +60,64 @@
     "phase": "liquid",
     "price": 2500,
     "freezing_point": 40
+  },
+  {
+    "id": "mana_potion",
```
Comment (ZhilkinSerg) [inline data/mods/Magiclysm/items/cast_spell_items.json]: You forgot to add `type` nodes here and below in this file.

**Pull request merged**